src/include/cell_inline.h

```diff
@@ -845,6 +848,27 @@ __wt_cell_unpack_safe(WT_SESSION_IMPL *session, const WT_PAGE_HEADER *dsk, WT_CE
         break;
     }
 
+    /*
+     * Unpack any fast-truncate information. Note that there is no way to write fast-truncate
+     * information to disk in versions before 6.1, but this information may still exist in the
+     * database files if we are downgrading from newer versions.
+     */
+    if (unpack->raw == WT_CELL_ADDR_DEL && F_ISSET(dsk, WT_PAGE_FT_UPDATE)) {
+        WT_RET(
+          __wt_vunpack_uint(&p, end == NULL ? 0 : WT_PTRDIFF(end, p), (uint64_t *)&page_del.txnid));
+        WT_RET(__wt_vunpack_uint(&p, end == NULL ? 0 : WT_PTRDIFF(end, p), &page_del.timestamp));
+        WT_RET(
+          __wt_vunpack_uint(&p, end == NULL ? 0 : WT_PTRDIFF(end, p), &page_del.durable_timestamp));
+        page_del.prepare_state = 0;            /* No prepare can have been in progress. */
+        page_del.previous_state = WT_REF_DISK; /* The leaf page is on disk. */
+        page_del.committed = 1;                /* There is no running transaction. */
+
+        /* Avoid a stale transaction ID on restart. */
+        if (dsk->write_gen <= S2BT(session)->base_write_gen &&
+          !F_ISSET(session, WT_SESSION_DEBUG_DO_NOT_CLEAR_TXN_ID))
+            page_del.txnid = WT_TXN_NONE;
```

Comment:
This code is no longer required as we are just unpacking the existing `page_del` info from the cell.

Reply:
I think we still need this information in the downgraded version. If the fast truncate is stable and visible, the downgraded version should be able to read that and return not found for any data in the subtree. I don't think we can directly ignore this information.

I assume we will always downgrade from a clean shut down. Therefore, there is no need to handle the fast truncate information in rts. But I think we better to still do that unless we have a way to prevent the user to directly downgrade from an unclean shutdown.

Reply:
@quchenhao My intention for this change was only to be able to read database files from newer versions which may or may not write the additional fast-truncate information. There would be no guarantee of whether all the fast-truncate behavior would be the same as newer versions.

Reply:
We should address the issue of a potential disk leak elsewhere, as it should also be backported to older versions and I feel that it is separate from this change. The disk leak should not be an issue currently as MongoDB does not write out fast-truncate information. I've created a new ticket WT-11321 to track the issue, and it should be completed before WT-10307 but doesn't need to be completed for this current ticket.